### ISSUE TITLE
Do not report SA1121 when type was aliased

### ISSUE THREAD

**Issue by froufrou121** [DotNetAnalyzers/StyleCopAnalyzers#2042]
In the following example StyleCop.Analyzers reports SA1121 wherever MY_DATA is used. This should not be the case as I decided to create the alias for a good reason.

namespace TestStyle
{
    using System.Collections.Generic;
    using MY_DATA = System.Int32;

    public class TestAlias
    {
        public IEnumerable<MY_DATA> Get()
        {
            return new MY_DATA[0];
        }
    }
}

Also note that you cannot write using MY_DATA = int; as this is not accepted by C#.

**Comment by sharwell** [comment]
This is currently by design, and we have a [whole series of tests](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/33d6540ee2fd7faa0e5aad19b87130e2d49ffe7c/StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1121UnitTests.cs#L771-L853) to make sure this warning will be reported. We also took a bunch of steps to make sure the test is efficient in #367/#967.

I'll leave the issue open for discussion, but right now I think this is the behavior we want, especially in light of the second manner of addressing this below.

There are three ways around this:

1. Use a suppression to hide the warning.

2. Create a wrapper structure which contains a single field. The JIT does a very good job of optimizing this to be just as efficient as your alias, but you actually get a strong type for it to avoid misuse and better communicate your API. For example:

    ```csharp
    public struct MY_DATA
    {
        private int value;
    }
    ```

3. If `MY_DATA` is a name based on a native API, use a `NativeMethods` class instead of the current `TestAlias` class.

**Comment by dlemstra** [comment]
I have a slightly different situation. Based on a conditional compilation directive a type is chosen:
```C#
#if Q8
using QuantumType = System.Byte;
#elif Q16
using QuantumType = System.UInt16;
#elif Q16HDRI
using QuantumType = System.Single;
#else
#error Not implemented!
#endif
````

None of the suggestion options are an option for me. Would it be possible to add a setting to the stylecop.json file where we can enable or disable the use of a type alias for this rule? I might be able to help out and send a pull request if adding this feature is an acceptable solution.

**Comment by bjornhellander** [comment]
I don't agree with ignoring aliases completely, but see #3640 for a different approach. Do you think that would be a good thing? I know this issue is old, but just in case you are still interested.

**Comment by Kevin-Seiden** [comment]
My two cents on this issue is that having the detection is not the problem, the problem is that there are two unique situations that aren't uniquely configurable.  SA1121 might be desirable in code but not in defined using aliases.

Considering the following:

```c#
using Cookie = System.Int32;
```

``` c#
const Cookie edibleCookie = 6; //SA1121 Detection 1

public void EatCookie(System.Int32 cookie) // SA1121 Detection 2
{
  if(cookie == edibleCookie)
  {
    // Eat it!
  }
}
```

In this case you have two instances of SA1121, Detection 1 is the one that is questionable, while Detection 2 is typically most desired.  Our only options are to
1. Disable SA1121 for all cases.
2. Suppress it at every single location it's used in a using alias, which is tedious.
3. Don't use using aliases, which is a useful feature for self documenting code.


Given the choice I'd disable it but then it won't identify cases like Detection 2 above.  Would it be possible to separate these as two different analyzer rules so we could configure it to our style standard?

All of this is further complicated by the fact that the C# 12 added this:
```c#
using Cookie = int; // But wait! I am using a built-in type alias!
```
